KarhooSDK/Service/Loyalty/KarhooLoyaltyService.swift

```diff
@@ -64,4 +70,13 @@ final class KarhooLoyaltyService: LoyaltyService {
         loyaltyPreAuthInteractor.set(loyaltyPreAuth: preAuthRequest)
         return Call(executable: loyaltyPreAuthInteractor)
     }
+    
+    func getCurrentLoyaltyStatus(identifier: String) -> LoyaltyStatus? {
+        return userDataStore.getLoyaltyStatusFor(loyaltyId: identifier)
```

Comment:
accessing the `userDataStore` can be done outside the `LoyaltyService`. in my undrestanding the service for our SDK is more about the network and API. 

If we need to cache it and read it from the cache, it can be done in a separate place.

Reply:
That's the case on Android, but not on iOS. On iOS the user store is internal and the services are public so getting information from there requires going through services.